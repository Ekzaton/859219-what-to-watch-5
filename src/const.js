export const APIRoute = {
  FILMS: `/films/`,
  FAVORITE: `/favorite/`,
  PROMO: `/films/promo/`,
  COMMENTS: `/comments/`,
  LOGIN: `/login/`
};

export const AppRoute = {
  ROOT: `/`,
  SIGN_IN: `/login/`,
  MY_LIST: `/mylist/`,
  REVIEW: `/review/`,
  REVIEW_ID: `/films/:id/review/`,
  PLAYER: `/player/`,
  PLAYER_ID: `/player/:id/`,
  FILMS: `/films/`,
  FILMS_ID: `/films/:id/`
};

export const Tab = {
  OVERVIEW: `overview`,
  DETAILS: `details`,
  REVIEWS: `reviews`
};

export const RatingLevel = {
  BAD: `Bad`,
  NORMAL: `Normal`,
  GOOD: `Good`,
  VERY_GOOD: `Very good`,
  AWESOME: `Awesome`,
  UNRATED: `Unrated`
};

export const Review = {
  MIN_LENGTH: 50,
  MAX_LENGTH: 400,
};

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const DEFAULT_GENRE = `All genres`;

export const DEFAULT_MOVIES_COUNT = 8;
