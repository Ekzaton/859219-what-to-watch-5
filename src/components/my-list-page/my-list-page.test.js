import React from "react";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import {Route, BrowserRouter} from "react-router-dom";

import {AuthorizationStatus} from "../../const";

import mockMovies from "../../mocks/movies";

import MyListPage from "./my-list-page";

describe(`MyListPage snapshot testing`, () => {
  it(`renders store-connected component correctly`, () => {
    const mockStore = configureStore([]);

    const store = mockStore({
      APP_DATA: {
        favoriteMovies: mockMovies,
      },
      APP_USER: {
        status: AuthorizationStatus.AUTH,
      },
    });

    const myListPageComponent = renderer.create(
        <Provider store={store}>
          <BrowserRouter>
            <Route>
              <MyListPage/>
            </Route>
          </BrowserRouter>
        </Provider>
    );

    expect(myListPageComponent.toJSON()).toMatchSnapshot();
  });
});
