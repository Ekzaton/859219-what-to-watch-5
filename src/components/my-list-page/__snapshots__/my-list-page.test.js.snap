// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyListPage snapshot testing renders store-connected component correctly 1`] = `
<div
  className="user-page"
>
  <header
    className="page-header user-page__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <h1
      className="page-title user-page__title"
    >
      My list
    </h1>
    <div
      className="user-block"
    >
      <div
        className="user-block__avatar"
      >
        <img
          alt="User avatar"
          height="63"
          src="/img/avatar.jpg"
          width="63"
        />
      </div>
    </div>
  </header>
  <section
    className="catalog"
  >
    <h2
      className="catalog__title visually-hidden"
    >
      Catalog
    </h2>
    <div
      className="catalog__movies-list"
    >
      <article
        className="small-movie-card catalog__movies-card"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <a
          className="small-movie-card__image"
          href="/films/1"
          onClick={[Function]}
        >
          <video
            height="175"
            muted={true}
            poster="https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg"
            src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
            width="280"
          />
        </a>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/1"
            onClick={[Function]}
          >
            Bronson
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <a
          className="small-movie-card__image"
          href="/films/2"
          onClick={[Function]}
        >
          <video
            height="175"
            muted={true}
            poster="https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/pulp-fiction.jpg"
            src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
            width="280"
          />
        </a>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/2"
            onClick={[Function]}
          >
            Pulp Fiction
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <a
          className="small-movie-card__image"
          href="/films/3"
          onClick={[Function]}
        >
          <video
            height="175"
            muted={true}
            poster="https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/midnight-special.jpg"
            src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
            width="280"
          />
        </a>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/3"
            onClick={[Function]}
          >
            Midnight Special
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <a
          className="small-movie-card__image"
          href="/films/4"
          onClick={[Function]}
        >
          <video
            height="175"
            muted={true}
            poster="https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/matrix.jpg"
            src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
            width="280"
          />
        </a>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/4"
            onClick={[Function]}
          >
            Matrix
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <a
          className="small-movie-card__image"
          href="/films/5"
          onClick={[Function]}
        >
          <video
            height="175"
            muted={true}
            poster="https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/legend.jpg"
            src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
            width="280"
          />
        </a>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/5"
            onClick={[Function]}
          >
            Legend
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <a
          className="small-movie-card__image"
          href="/films/6"
          onClick={[Function]}
        >
          <video
            height="175"
            muted={true}
            poster="https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/gangs_of_new_york.jpg"
            src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
            width="280"
          />
        </a>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/6"
            onClick={[Function]}
          >
            Gangs of new york
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <a
          className="small-movie-card__image"
          href="/films/7"
          onClick={[Function]}
        >
          <video
            height="175"
            muted={true}
            poster="https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/A_Star_Is_Born.jpg"
            src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
            width="280"
          />
        </a>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/7"
            onClick={[Function]}
          >
            A Star Is Born
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <a
          className="small-movie-card__image"
          href="/films/8"
          onClick={[Function]}
        >
          <video
            height="175"
            muted={true}
            poster="https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/beach.jpg"
            src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
            width="280"
          />
        </a>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/8"
            onClick={[Function]}
          >
            Beach
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <a
          className="small-movie-card__image"
          href="/films/9"
          onClick={[Function]}
        >
          <video
            height="175"
            muted={true}
            poster="https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/aviator.jpg"
            src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
            width="280"
          />
        </a>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/9"
            onClick={[Function]}
          >
            Aviator
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <a
          className="small-movie-card__image"
          href="/films/10"
          onClick={[Function]}
        >
          <video
            height="175"
            muted={true}
            poster="https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/johnny-english.jpg"
            src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
            width="280"
          />
        </a>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/10"
            onClick={[Function]}
          >
            Johnny English
          </a>
        </h3>
      </article>
    </div>
  </section>
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="copyright"
    >
      <p>
        © 2019 What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;
